src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -235,8 +235,8 @@ private BufferSegment AllocateSegment(int sizeHint)
             }
             else
             {
-                // We can't use the pool so allocate an array
-                newSegment.SetUnownedMemory(new byte[sizeHint]);
+                // We can't use the pool so use the array pool
+                newSegment.SetOwnedMemory(ArrayPool<byte>.Shared.Rent(sizeHint));
```

Comment:
Why is this case different from the `_pool is null` case? i.e. why does that one use GetSegmentSize and this one doesn't (in which case the conditions could also be combined)?  Same question for the writer below.

Reply:
Yeah should be able to